docs/refs/api/rtos/callbacks.md

```diff
@@ -169,25 +70,13 @@ int main() {
 }
 ```
 
-#### Combinatorial explosion damage control (aka why a separate class for Callbacks)
+#### A separate class for Callbacks)
 
-Unfortunately, supporting all of the standard C++ function types is difficult.
+C++ requires a large set of overloads to support all of the standard function types. It is unreasonable to expect a new library author to add all of these overloads to every function that could take in a callback.
 
-1. State is important, so need to support either C-style function pointers with state, or C++ member function pointers.
+However, C++ provides the tools to delegate this problem to a single class. This class is the Callback class. The Callback class should be familiar to users of the `std::function` class that C++11 introduced but is available for older versions of C++.
```

Comment:
A bit ambiguous, how about:
The Callback class should be familiar to users of the `std::function` class that C++11 introduced but the Callback class is available for older versions of C++